rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -566,6 +566,17 @@ def parse_primitive_value_string(type_, value_string):
         for quote in ['"', "'"]:
             if value_string.startswith(quote) and value_string.endswith(quote):
                 value_string = value_string[1:-1]
+                if quote == '"':
+                    match = re.search(r'(?<!\\)"', value_string)
+                else:
+                    match = re.search(r"(?<!\\)'", value_string)
+                if match is None:
+                    value_string = value_string.replace('\\' + quote, quote)
+                else:
+                    raise InvalidValue(
+                        primitive_type,
+                        value_string,
+                        'string inner quotes not properly escaped')
```

Comment:
If you invert the condition (`if match is not None: raise ...`) you don't need an else block.